### DISCUSSION TITLE
Issue with setModelLanguage deadlocking browser tab

### DISCUSSION THREAD

**Discussion by tankbusta** [CodinGame/monaco-vscode-api#474] [Q&A] [answered]
Hey folks,

I'm not sure what I'm doing wrong here but I have a basic Vue component I'm working on. It's supposed to be a lightweight editor with language support and a dropdown to change supported language. Whenever the user selects a new language from the deadline, it spins the CPU for 100%. 

Any idea what I'm doing wrong/missing to cause this?

Thanks!

Repo: https://github.com/tankbusta/monaco-vscode-hang/tree/main

**Comment by CGNonofr** [comment]
I'm not very familiar with vue, are you sure there is no infinite loop in your code?

Btw, reusing the same model for multiple languages is kind of weird

**Reply by tankbusta** [reply]
I'm pretty sure it's not a vue problem, my current thought is that it's a problem with the vscode extension host but im not sure why.


When you say using the same model for multiple languages is weird. Are you supposed to create a model for the same file every time the language is changed? The dropdown I have is supposed to imitate the language selection in the status bar that I don't have. If the answer is yes, then maybe that's the cause because if you start off on a language it works fine but switching causes the deadlock.

**Reply by CGNonofr** [reply]
Updating a model language shouldn't be a problem though.

> Are you supposed to create a model for the same file every time the language is changed?

A model is link to a file, a file has an extension, and extension is related to a language. I don't really see the point of reusing the same model

**Reply by tankbusta** [reply]
I've tried a few other changes; looking at the Demo to try and figure out what I'm doing wrong with no luck. I'm trying a new approach to use the native language selector in VSCode but it doesn't seem to switch when you use the "Change Language Mode" option. Of note, I had to add `.lua` to the file being created otherwise it'd be stuck in "plaintext" mode. 

It feels like I'm potentially missing a service override but I'm not familiar with VSCode/Monaco to know which one I'm missing.

![image](https://github.com/user-attachments/assets/29c1b142-abd9-4b8c-831d-fbda44297c60)
Code: https://github.com/tankbusta/monaco-vscode-hang/tree/without_lang_selector

**Reply by CGNonofr** [reply, marked as answer]
I'm not familiar with vue; but the issue seems to come from vue creating proxy for refs

using `const editor = toRaw(unref(editorRef));` instead of `const editor = unref(editorRef);` seems to fix the issue

**Reply by tankbusta** [reply]
Wow, it really was Vue! I'm honestly not too familiar with proxies so I'll do some additional reading but I really appreciate the guidance.
